File: POOBanca/C++/Revista.cpp
#include "Revista.h"

void Revista::apresentacao() {
    Produto::apresentacao();

    cout << "O número da revista é " << this->numero << "." << endl;
}

Revista::Revista(string nome, string editora, int totPaginas, int numero) : Produto(nome, editora, totPaginas) {
    this->numero = numero;
}

Revista::~Revista() {
    cout << "Objeto Revista Destruído com Sucesso!" << endl;
}

int Revista::getNumero() {
    return this->numero;
}

void Revista::setNumero(int numero) {
    this->numero = numero;
}

File: POOBanca/Python/Revista.py
from Produto import Produto

class Revista(Produto):
    __numero = 0

    def apresentacao(self):
        super().apresentacao()

        print("O número da revista é {}.".format(self.__numero))

    def __init__(self, nome, editora, totPaginas, numero):
        super().__init__(nome, editora, totPaginas)

        self.__numero = numero

    def __del__(self):
        print("Objeto Revista Destruído com Sucesso!")

    def getNumero(self):
        return self.__numero

    def setNumero(self, numero):
        self.__numero = numero